lib/cfn-nag/custom_rules/IamUserLoginProfilePasswordResetRule.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+require "cfn-nag/violation"
+require_relative "base"
+
+class IamUserLoginProfilePasswordResetRule < BaseRule
+  def rule_text
+    "IAM User Login Profile should exist and have PasswordResetRequired property set to true"
+  end
+
+  def rule_type
+    Violation::WARNING
+  end
+
+  def rule_id
+    "W50"
+  end
+
+  def audit_impl(cfn_model)
+    violating_iam_users = cfn_model.resources_by_type("AWS::IAM::User").select do |iam_user|
+      iam_user.loginProfile["PasswordResetRequired"].nil? || iam_user.loginProfile["PasswordResetRequired"].to_s == "false"
```

Comment:
make sure to use the proper truthy? predicate here.  cfn is "weakly typed" when it comes to boolean values

a few things:
1. this code is going to fail when loginProfile isn't there.  it's not a required field
2. i'm not sure what the proper behavior is here.... i guess if there is a login profile, the password should be noecho and reset must be required?